frontend/src/__tests__/cypress/cypress/e2e/projects/Permissions.cy.ts

```diff
@@ -0,0 +1,315 @@
+      groupTable.findTableHeaderButton('Permission').should(be.sortAscending);
+      groupTable.findTableHeaderButton('Permission').click();
+      groupTable.findTableHeaderButton('Permission').should(be.sortDescending);
+
+      groupTable.findTableHeaderButton('Date added').click();
+      groupTable.findTableHeaderButton('Date added').should(be.sortAscending);
+      groupTable.findTableHeaderButton('Date added').click();
+      groupTable.findTableHeaderButton('Date added').should(be.sortDescending);
+    });
+
+    it('Add group', () => {
+      initIntercepts({ isEmpty: true });
+      cy.intercept(
+        {
+          method: 'POST',
+          pathname:
+            '/api/k8s/apis/rbac.authorization.k8s.io/v1/namespaces/test-project/rolebindings',
+        },
+        mockRoleBindingK8sResource({}),
+      ).as('addGroup');
+      permissions.visit('test-project');
+
+      permissions.findAddGroupButton().click();
+      groupTable.findAddInput().fill('group-1');
+      groupTable.selectPermission('group-1', 'Admin Edit the project and manage user access');
+      groupTable.findSaveButton('Group').click();
+
+      cy.wait('@addGroup').then((inerception) => {
+        expect(inerception.request.body).to.containSubset({
+          metadata: {
+            labels: {
+              'opendatahub.io/dashboard': 'true',
+              'opendatahub.io/project-sharing': 'true',
+            },
+          },
+          roleRef: { apiGroup: 'rbac.authorization.k8s.io', kind: 'ClusterRole', name: 'admin' },
+          subjects: [{ apiGroup: 'rbac.authorization.k8s.io', kind: 'Group', name: 'group-1' }],
+        });
+      });
+    });
+
+    it('Edit group', () => {
+      initIntercepts({ isEmpty: false });
+
+      cy.intercept(
+        {
+          method: 'POST',
+          pathname:
+            '/api/k8s/apis/rbac.authorization.k8s.io/v1/namespaces/test-project/rolebindings',
+        },
+        mockRoleBindingK8sResource({}),
+      ).as('editGroup');
+      cy.intercept(
+        {
+          method: 'DELETE',
+          pathname:
+            '/api/k8s/apis/rbac.authorization.k8s.io/v1/namespaces/test-project/rolebindings/group-1',
+        },
+        { kind: 'Status', apiVersion: 'v1', metadata: {}, status: 'Success' },
+      ).as('deleteGroup');
+
+      permissions.visit('test-project');
+
+      groupTable.getTableRow('group-1').findKebabAction('Edit').click();
+      groupTable.findEditInput('group-1').clear().type('group-3');
+      groupTable.selectPermission('group-3', 'Admin Edit the project and manage user access');
+      groupTable.findSaveButton('group-3').click();
+
+      cy.wait('@editGroup').then((interception) => {
+        expect(interception.request.body).to.containSubset({
+          metadata: {
+            labels: {
+              'opendatahub.io/dashboard': 'true',
+              'opendatahub.io/project-sharing': 'true',
+            },
+          },
```

Comment:
You should include the metadata name here because it's important that the edit attempts to change the same resource.

Reply:
Added an assertion for metadata name, as its value has random characters.

Reply:
@ppadti why would the edit scenario have random values? isn't it just editing the the resource that was provided on load which is something we control from the cypress mocks?

Reply:
Oh no!  My mistake.
I thought this was a `PUT` request. But it's a `POST` request, which is _creation_. That's certainly odd. Looks like editing does a create and a delete...
Ok now my request to assert the name for _edit_ is invalid because we're not editing a previous resource.

Reply:
yes, the edit function will delete the existing one and create new one.
Okay then I will remove that.

Reply:
Updated the PR.